libs/lz4/src/test/java/org/elasticsearch/lz4/ESLZ4DecompressorTests.java

```diff
@@ -36,4 +38,27 @@ public void testDecompress() {
             assertArrayEquals(uncompressed, output);
         }
     }
+
+    public void testDecompressRandomBytes() throws IOException {
+        for (int i = 0; i < 15; ++i) {
+            int uncompressedBytesLength = randomFrom(16, 32, 64, 128, 256, 512, 1024) * 1024;
+
+            BytesStreamOutput bytesStreamOutput = new BytesStreamOutput(uncompressedBytesLength);
+            for (int j = 0; j < uncompressedBytesLength / 4; ++j) {
+                bytesStreamOutput.writeInt(randomFrom(0, 1, randomInt()));
+            }
+            byte[] uncompressed = new byte[uncompressedBytesLength];
+            bytesStreamOutput.bytes().streamInput().read(uncompressed);
+
+            byte[] compressed = new byte[uncompressed.length + uncompressed.length / 255 + 16];
+            LZ4Compressor compressor = LZ4Factory.safeInstance().fastCompressor();
+            compressor.compress(uncompressed, compressed);
+
+            LZ4FastDecompressor decompressor = ESLZ4Decompressor.INSTANCE;
+            byte[] output = new byte[uncompressed.length];
+            decompressor.decompress(compressed, output);
+
+            assertArrayEquals(uncompressed, output);
```

Comment:
Given that the test in `ESLZ4CompressorTests` now verifies that the compression gives same results, perhaps move this there too?